src/InterfaceLimits.jl

```diff
@@ -114,87 +167,155 @@ end
 function add_injector_constraint!( # for buses that have loads and generators
     m::Model,
     injector_type::Type{StaticInjection},
-    p_var; #injection variable,
+    p_var, #injection variable,
+    hvdc_inj;
     ldf_lim = nothing,
     max_gen = nothing, # pass this if enforce_gen_limits = true
 )
     if !isnothing(ldf_lim)
-        @constraint(m, p_var >= ldf_lim)
+        @constraint(m, p_var - hvdc_inj >= ldf_lim)
         !isnothing(max_gen) && (max_gen += ldf_lim)
     end
-    !isnothing(max_gen) && @constraint(m, p_var <= max_gen)
+    !isnothing(max_gen) && @constraint(m, p_var - hvdc_inj <= max_gen)
 end
 
 function add_injector_constraint!( # for buses that have loads only
     m::Model,
     injector_type::Type{ElectricLoad},
-    p_var; #injection variable,
+    p_var, #injection variable,
+    hvdc_inj;
     ldf_lim = nothing,
     max_gen = nothing, # pass this if enforce_gen_limits = true
 )
     if !isnothing(ldf_lim)
-        @constraint(m, p_var == ldf_lim)
+        @constraint(m, p_var - hvdc_inj == ldf_lim)
     else
-        @constraint(m, p_var <= 0.0)
+        @constraint(m, p_var - hvdc_inj <= 0.0)
     end
+
 end
 
 function add_injector_constraint!( # for buses that have generators only
     m::Model,
     injector_type::Type{Generator},
-    p_var; #injection variable,
+    p_var, #injection variable,
+    hvdc_inj;
+    ldf_lim = nothing,
+    max_gen = nothing, # pass this if enforce_gen_limits = true
+)
+    @constraint(m, p_var - hvdc_inj >= 0.0)
+    !isnothing(max_gen) && @constraint(m, p_var - hvdc_inj <= max_gen)
+end
+
+function add_injector_constraint!( # for TwoTerminalHVDCLine buses with no gens or loads
+    m::Model,
+    injector_type::Type{InterconnectingConverter},
+    p_var, #injection variable,
+    hvdc_inj;
     ldf_lim = nothing,
     max_gen = nothing, # pass this if enforce_gen_limits = true
 )
-    @constraint(m, p_var >= 0.0)
-    !isnothing(max_gen) && @constraint(m, p_var <= max_gen)
+    @constraint(m, p_var - hvdc_inj == 0.0)
 end
 
 function add_variables!(
-    m,
-    inames,
-    in_branches,
-    gen_buses,
-    load_buses,
-    security,
-    c_branches, #must not be nothing if security = true
-    enforce_load_distribution,
+    m::Model,
+    inames::Vector{String},
+    in_branches::Vector{ACBranch},
+    gen_buses::Set,#PSY.FlattenIteratorWrapper{Bus},
+    load_buses::Set,# PSY.FlattenIteratorWrapper{Bus},
+    hvdc_buses::Set,# PSY.FlattenIteratorWrapper{Bus},
+    security::Union{Bool, Security},
+    enforce_load_distribution::Bool,
 )
     # create flow variables for branches
     @variable(m, F[inames, get_name.(in_branches)])
     @variable(m, I[inames])
-    @variable(m, P[inames, get_name.(union(gen_buses, load_buses))])
+    @variable(m, P[inames, get_name.(union(gen_buses, load_buses, hvdc_buses))])
```

Comment:
I added hvdc_buses to the union as well to make sure buses at either end of TwoTerminalHVDCLines are added to the list of injection variables even if there are no other loads or generators at those buses.